src/cljs_rest/core.cljs

```diff
@@ -159,11 +159,38 @@
 ;; Listing
 
 (defprotocol RestfulListing
+  (first-item [_] [_ params])
   (item-constructor [_])
   (items-constructor [_]))
 
 (defrecord ResourceListing [url opts item-opts constructor ok? data]
   RestfulListing
+  (first-item [this] (first-item this nil))
+
+  (first-item [this params]
+    (go
+      (let [resources (<! (read this params))
```

Comment:
`resources` is really the response right? Would it be more clear here and below named `response`?